test/model/message_list_test.dart

```diff
@@ -162,5 +162,107 @@ void main() async {
     test('rendering-only update does not change timestamp (for old server versions)', () async {
       await checkRenderingOnly(legacy: true);
     });
+
+    group('ReactionEvent handling', () {
+      ReactionEvent mkEvent(Reaction reaction, ReactionOp op, int messageId) {
+        return ReactionEvent(
+          id: 1,
+          op: op,
+          emojiName: reaction.emojiName,
+          emojiCode: reaction.emojiCode,
+          reactionType: reaction.reactionType,
+          userId: reaction.userId,
+          messageId: messageId,
+        );
+      }
+
+      test('add reaction', () async {
+        final originalMessage = eg.streamMessage(id: 243, stream: stream, reactions: []);
+        final messageList = await messageListViewWithMessages([originalMessage], stream, narrow);
+
+        final message = messageList.messages.single;
+        check(message).reactions.not(it()..jsonEquals([eg.unicodeEmojiReaction]));
```

Comment:
I think this more direct version would be fine here:
```suggestion
        check(message).reactions.jsonEquals([]);
```

The reader can compare that with the `jsonEquals` below and see that the expected values are clearly different — one's empty and one's a singleton.